sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -112,7 +112,7 @@ public synchronized void setMetrics(
 
     try {
       // NOTE: removing an activity does not reset the frame counts, only reset() does
-      frameMetricsAggregator.remove(activity);
+      activity.runOnUiThread(() -> frameMetricsAggregator.remove(activity));
```

Comment:
```suggestion
      activity.runOnUiThread(() -> {
          try {
              frameMetricsAggregator.remove(activity);
          } catch (Throwable ignored) {
          // throws IllegalArgumentException when attempting to remove OnFrameMetricsAvailableListener
          // that was never added.
          // there's no contains method.
          // throws NullPointerException when attempting to remove OnFrameMetricsAvailableListener and
          // there was no
          // Observers, See
          // https://android.googlesource.com/platform/frameworks/base/+/140ff5ea8e2d99edc3fbe63a43239e459334c76b
    }
      );
```

Reply:
In fact, we don't need the outer catch block then (unless runOnUiThread might blow up for some reason)